src/js/components/Calendar/__tests__/Calendar-test.tsx

```diff
@@ -252,6 +252,27 @@ describe('Calendar', () => {
     expect(container.firstChild).toMatchSnapshot();
   });
 
+  test('read-only with range when no onSelect prop provided', () => {
+    const { getByText } = render(
+      <Grommet>
+        <Calendar date={DATE} range animate={false} />
+      </Grommet>,
+    );
+
+    // Verify calendar renders without throwing
+    expect(getByText('17')).toBeInTheDocument();
+    expect(getByText('20')).toBeInTheDocument();
+
+    // Click multiple dates - should not cause any errors
+    // Since there's no onSelect, range selection should not work
+    fireEvent.click(getByText('17'));
+    fireEvent.click(getByText('20'));
+
+    // Calendar should still be accessible after clicks
+    expect(getByText('17')).toBeInTheDocument();
+    expect(getByText('20')).toBeInTheDocument();
```

Comment:
I think this test would pass even before the changes from https://github.com/grommet/grommet/pull/7830 were merged

Reply:
fixed